Add explanations

dviererbe opened the pull request:
Finally I can land this set of changes. It contains explanations, numerous new glossary entries (and finally the first definitions) and some format refreshing.

If you want to laugh a bit; do yourself a favor and check out the glossary entry for "Circle of Friends". This picture was not easy to find. ^^

P.S. It is not "today" in my timezone, but in UTC it is still "today". :D

dviererbe committed:
feat: Add globally shared links

The docs/reuse/links.rst can be used to share links between documents.

dviererbe committed:
fix: sphinx warning "glossary terms must not be separated by empty lines"

dviererbe committed:
refactor: replaced WIP banner in every article with rst_prolog in conf.py

dviererbe committed:
refactor: unify title heading format and remove unused refs

dviererbe committed:
chore(glossary): remove unused terms 'Subversion' and 'SVN'

dviererbe committed:
feat(glossary): Add images for term "Circle of Friends"

dviererbe committed:
feat: Add article reference/architectures

Also add the terms:
 * Architecture
 * Big-Endian
 * Central Processing Unit
 * CISC
 * Complex Instruction Set
 * CPU
 * Endianness
 * Little-Endian
 * Reduced Instruction Set
 * RISC

dviererbe committed:
feat(glossary): Add term '80x86'

dviererbe committed:
feat(glossary): Add term 'AArch32'

dviererbe committed:
feat(glossary): Add term 'AArch64'

dviererbe committed:
feat(glossary): Add definition for 'Application Programming Interface' and 'Application Binary Interface'

Also adds warnings to not confuse these terms with each other.

dviererbe committed:
feat(glossary): Add definition for term 'amd64'

Also adds related terms:
 * Intel 64
 * Intel x86
 * x64
 * x86
 * x86-64
 * x86_64

dviererbe committed:
feat(glossary): Add term 'Source Code'

dviererbe committed:
feat(glossary): Add term 'Whitespace'

dviererbe committed:
feat(glossary): Add term 'Stack'

dviererbe committed:
feat(glossary): Add term 'Staging Environment'

dviererbe committed:
feat(glossary): Add term 'Request for Comments'

dviererbe committed:
feat(glossary): Add term 'RFC'

dviererbe committed:
feat(glossary): Add term 'Perl Compatible Regular Expressions'

dviererbe committed:
feat(glossary): Add term 'PCRE'

dviererbe committed:
feat(glossary): Add term 'Regular Expression'

dviererbe committed:
feat(glossary): Add term 'RegEx'

dviererbe committed:
feat(glossary): Add term 'Repository'

dviererbe committed:
style: Change formatting of PKCS Wikipedia link

dviererbe committed:
feat(glossary): Add term 'Package Manager'

dviererbe committed:
feat(glossary): Add definition for term 'Operating System'

dviererbe committed:
feat(glossary): Add term 'National Institute of Standards and Technology'

dviererbe committed:
feat(glossary): Add term 'NIST'

dviererbe committed:
feat(glossary): Add term 'Merge Conflict'

dviererbe committed:
feat(glossary): Add term 'Mailing List'

dviererbe committed:
feat(glossary): Add term 'Kernel'

dviererbe committed:
feat(glossary): Add term 'ISO'

dviererbe committed:
style: reformated IBM homepage link

dviererbe committed:
chore(glossary): remove terms 'GDPR' and 'General Data Protection Regulation'

dviererbe committed:
feat(glossary): Add term 'Free Software'

dviererbe committed:
style: reformat 'Free and Open Source Software' Wikipedia link

dviererbe committed:
feat(glossary): Add term 'Fork'

dviererbe committed:
feat(glossary): Add term 'Ubuntu Desktop'

dviererbe committed:
style: reformat Ubuntu Pro homepage link

dviererbe committed:
feat(glossary): Add term 'Ubuntu Server'

dviererbe committed:
style: reformat IRC Council (Ubuntu Wiki) link

dviererbe committed:
feat(glossary): Add definition for Ubuntu Developer Summit and Ubuntu Summit

dviererbe committed:
feat(glossary): Add term 'Ubuntu Code of Conduct'

dviererbe committed:
style: reformat Cloud Archive (Ubuntu Wiki) link

dviererbe committed:
feat(glossary): Add definition for term 'tarball'

dviererbe committed:
feat(glossary): Add term 'Distribution'

dviererbe committed:
feat(glossary): Add definition for term 'diff'

dviererbe committed:
style: reformat Developer Membership Board (Ubuntu Wiki) link

dviererbe committed:
feat(glossary): Add ARM terms

dviererbe committed:
feat(glossary): Add term 'Embedded Systems'

dviererbe committed:
feat(glossary): Add term 'Domain Name System'

dviererbe committed:
feat(glossary): Add term 'DNS'

dviererbe committed:
feat(glossary): Add definition for term 'Debian'

dviererbe committed:
feat(glossary): Add definition for term 'Codename'

dviererbe committed:
feat(glossary): Add definition for term 'Closed Source Software'

dviererbe committed:
feat(glossary): Add term 'Canonical'

dviererbe committed:
feat(glossary): Add definition for term 'Bug Tracking System'

dviererbe committed:
feat(glossary): Add definition for term 'Bug'

dviererbe committed:
feat(glossary): Add definition for term 'Breezy'

dviererbe committed:
feat(glossary): Add term 'Binary Package'

dviererbe committed:
feat(glossary): Add definition for term 'Bazaar'

dviererbe committed:
feat(glossary): Add term 'Source Package'

dviererbe committed:
feat(glossary): Add definition for term 'Rules File'

dviererbe committed:
feat(glossary): Add definition for term 'Watch File'

dviererbe committed:
feat(glossary): Add definition for term 'Control File'

dviererbe committed:
feat(glossary): Add definition for term 'Copyright File'

dviererbe committed:
feat(glossary): Add definition for term 'Changelog'

dviererbe committed:
chore(glossary): Add minor cross-ref to term

dviererbe committed:
feat(glossary): Add definition for term 'Version Control System'

dviererbe committed:
feat(glossary): Add term 'User Experience'

dviererbe committed:
feat(glossary): Add term 'User Interface'

dviererbe committed:
feat(glossary): Add term 'UX'

dviererbe committed:
feat(glossary): Add term 'Uniform Resource Identifier'

dviererbe committed:
feat(glossary): Add term 'Uniform Resource Locator'

dviererbe committed:
feat(glossary): Add definition for term 'Unix'

dviererbe committed:
feat(glossary): Add term 'URI'

dviererbe committed:
feat(glossary): Add term 'URL'

dviererbe committed:
feat(glossary): Add definition for term 'Patch'

dviererbe committed:
feat(glossary): Add term 'original tarball'

dviererbe committed:
rename 'Master of the Universe' to 'Masters of the Universe'

dviererbe committed:
feat(glossary): Add definition for term 'Image'

dviererbe committed:
feat(glossary): Add definition for term 'i386'

dviererbe committed:
feat(glossary): Add definition for term 'GNU'

dviererbe committed:
feat(glossary): Add definition for term 'Federal Information Processing Standards'

dviererbe committed:
style: reformat Expanded Security Maintenance (homepage) link

dviererbe committed:
feat(glossary): Add definition for term 'Code Review'

dviererbe committed:
chore: mention Ubuntu CoC for term Code of Conduct

dviererbe committed:
feat(glossary): Add term 'deb' and 'debs'

dviererbe committed:
fix(glossary): term DNS was in the wrong location

dviererbe committed:
feat(glossary): Add term 'dsc'

dviererbe committed:
feat(glossary): Add definition for term 'End of Life'

dviererbe committed:
feat: Add content for explanation/upstream-and-downstream

dviererbe committed:
feat: Add content for explanation/package-model

dviererbe committed:
refactor: rename explanation/ubuntu-release-cadence to explanation/releases

dviererbe committed:
chore: split explanation/launchpad-and-archive

into:
- explanation/archive
- explanation/development-process
- explanation/launchpad

dviererbe committed:
chore: reorder explanation toc tree

dviererbe committed:
feat: Add article explanation/main-inclusion-review

dviererbe committed:
chore(glossary): Add minor cross-reference

dviererbe committed:
feat(glossary): Add term 'Current Release in Development'

dviererbe committed:
feat(glossary): Add term 'Devel'

dviererbe committed:
feat(glossary): Add defition for term 'Downstream'

dviererbe committed:
feat(glossary): Add defition for term 'End of Line'

dviererbe committed:
feat(glossary): Add defition for term 'Launchpad'

dviererbe committed:
feat(glossary): Add defition for term 'Linux'

dviererbe committed:
feat(glossary): Add term 'Mirror'

dviererbe committed:
feat(glossary): Add term 'CD Mirror'

dviererbe committed:
chore: Add minor cross reference

dviererbe committed:
feat(glossary): Add defition for term 'Upstream'

dviererbe committed:
chore: Add minor cross reference

dviererbe committed:
feat(glossary): Add term 'Ubuntu autopkgtest Cloud'

dviererbe committed:
feat(glossary): Add defition for term 'Ubuntu'

dviererbe committed:
chore: Add minor cross-refernce

dviererbe committed:
feat(glossary): Add term 'Native Package'

dviererbe committed:
feat(glossary): Add defition for term 'Archive'

dviererbe committed:
feat: Add content for article explanation/archive

dviererbe committed:
feat: Add content for article explanation/launchpad

dviererbe committed:
feat: Add content for article explanation/releases

dviererbe committed:
style: Set table headings to aligns left

dviererbe committed:
chore: remove reference/source-package-formats

dviererbe committed:
chore: Refresh how-to/get-package-source format

Apply the styling that was used in other articles.

dviererbe committed:
chore: Refresh how-to/install-built-packages format

Apply the styling that was used in other articles.

dviererbe committed:
chore: Add minor cross-references

dviererbe committed:
feat(glossary): Add defition for term 'Archive Admin'

s-makin commented on docs/explanation/archive.rst:
```diff
@@ -0,0 +1,348 @@
+has several :ref:`Repositories` that anyone can access. The *Ubuntu Package Archive* 
+hosts :term:`Debian` :term:`Binary Packages <Binary Package>` (``.deb`` files) and
+:term:`Source Packages <Source Package>` (``.dsc`` files). On :term:`Ubuntu` installations,
+the *Ubuntu Package Archive* is configured as the default source for the :term:`APT`
```
Should "APT package manager" be defined as a single term rather than two?

dviererbe replied:
I think here is a greater discussion that I would like for you and @keirthana to decide. 
When does it make sense to reference a term? I initially wanted a Wikipedia style page where a term is referenced, every time it occurs, but I agree that this can be hard to read.

s-makin replied:
Wikipedia only tends to reference the first appearance of a term, I think :thinking: 
I'd tend to overlook a second reference if it occurred further down the page (especially if the page is long or if it makes sense that readers might want to remind themselves of the meaning at that point) since Wikipedia sometimes also does this, but it should be the exception rather than the rule.

dviererbe replied:
Resolved with 6cbf3120a572322bf619fd3c28f2f6d4075b21ba

s-makin commented on docs/explanation/archive.rst:
```diff
@@ -0,0 +1,348 @@
+| Before the initial release of an :term:`Ubuntu` :ref:`ArchiveSeries`, this
+  *pocket* contains non-security-related updates to :term:`Packages <Package>`
+  in the :ref:`ArchivePockets_Release` *pocket* before they get uploaded to the
+  :ref:`ArchivePockets_Release` *pocket*.
```
Suggest re-ordering to before-then-after, and changing to bullet points since the pipe doesn't seem to do anything in terms of rendering in RTD. 

Also just want to clarify, because it's a bit confusing. With regard to the "before" point, you have referred twice to the release pocket - is that intentional? (the other has release + updates)

dviererbe replied:
The reordering is fine. I started with after, because this is the more common situation. An LTS series has a lifespan of up to 10 years and Interim series of up to to nine months, but they are just in development for 6 months.

dviererbe replied:
The pipes actually ensures that these 2 paragraphs start on their own line without having a gap between them.

So rather than 
```
After <content>. Before <content>.
```

or

```
After <content>.

Before <content>.
```

we get 

```
After <content>.
Before <content>.
```

, but bullet points are probably okay.

s-makin commented on docs/explanation/archive.rst:
```diff
@@ -0,0 +1,348 @@
+Components
+----------
+
+*Components* are logical subdivisions or namespaces of the :term:`Packages <Package>`
```
Do we have a definition for "namespaces"?

dviererbe replied:
namespaces is used here as an abstract concept rather than a concrete technical reference Many programmers know namespaces from their programming languages like C++/C# or Java/Python/JS/TS modules.

Do you think this should have a glossary entry?

s-makin replied:
Hmmm...maybe? If it's a common term that anyone getting involved in packaging should already know, then possibly not. I'll leave it to your judgement.

dviererbe replied:
Resolved with e23ad4092120f6d5ff3f0c61b6769ca1dd845016, I added a term definition for namespaces

s-makin commented on docs/explanation/archive.rst:
```diff
@@ -0,0 +1,348 @@
+subscribe to the *components* of a :ref:`ArchiveSuite`.
+
+The :term:`Packages <Package>` of an :term:`Ubuntu` :ref:`ArchiveSeries` are categorized
+if they are :term:`Open Source Software` and part of the Base :term:`Packages <Package>`
```
It seems like we maybe want to define "base packages"

dviererbe replied:
The components section was one of the harder sections to write. "base packages" was the term used here: https://ubuntu.com/about/release-cycle

There is not really a good definition. It's basically just the packages that Canonical decided to support, because they are fundamental for Ubuntu.

s-makin replied:
I think adding that should actually be enough to make it clear, i.e. "base packages, which are fundamental for Ubuntu". The reason I don't like "base" is because it has multiple potential meanings and so has the potential to be ambiguous if we haven't defined it, but adding the "fundamental" part sort of removes that ambiguity (imo).

dviererbe replied:
Resolved with eba8fc32ad870061b15e2174da97f7c800ea1ea9, I added a term definition

s-makin commented on docs/explanation/archive.rst:
```diff
@@ -0,0 +1,348 @@
+            fi.archive.ubuntu.com.	332	IN	CNAME	mirrors.nic.funet.fi.
+            mirrors.nic.funet.fi.	332	IN	A	193.166.3.5
+
+        Therefore ``mirrors.nic.funet.fi`` is finlands *Country Mirror*.
```
```suggestion
        Therefore, ``mirrors.nic.funet.fi`` is Finland's country mirror.
```

s-makin commented on docs/explanation/archive.rst:
```diff
@@ -0,0 +1,348 @@
+
+    .. tab-item:: Tunisia (TN)
+
+        Tunisia does not have any third-party mirrors in its country. Therefore the
+        Tunisia *Country Mirror* is just the main *Ubuntu Package Archive* server
```
```suggestion
        Tunisia does not have any third-party mirrors in its country. Therefore the
        Tunisia country mirror is just the main Ubuntu package archive server
```

s-makin replied:
I'd avoid use of the word "main" here to avoid new readers getting confused with the previous discussion of the *main* repository. Primary might work instead?

s-makin commented on docs/explanation/archive.rst:
```diff
@@ -0,0 +1,348 @@
+            archive.ubuntu.com.	1	IN	A	91.189.91.81
+            archive.ubuntu.com.	1	IN	A	91.189.91.82
+
+Package Uploads
```
```suggestion
Package uploads
```

s-makin commented on docs/explanation/archive.rst:
```diff
@@ -0,0 +1,348 @@
+Package Uploads
+---------------
+
+:term:`Ubuntu` encourages contributions from any person in the wider community.
```
```suggestion
Ubuntu encourages contributions from any person in the wider community.
```

s-makin commented on docs/explanation/archive.rst:
```diff
@@ -0,0 +1,348 @@
+---------------
+
+:term:`Ubuntu` encourages contributions from any person in the wider community.
+However, direct uploading to the *Ubuntu Package Archive* is restricted. These
```
```suggestion
However, direct uploading to the Ubuntu package archive is restricted. These
```

s-makin commented on docs/explanation/archive.rst:
```diff
@@ -0,0 +1,348 @@
+However, direct uploading to the *Ubuntu Package Archive* is restricted. These
+general contributions need to be reviewed and uploaded by a :term:`Sponsor`.
+
+See the article :doc:`/explanation/sponsoring` that explains this process in more detail.
```
```suggestion
See our :doc:`article on sponsoring <sponsoring>` that explains this process in more detail.
```

s-makin commented on docs/explanation/archive.rst:
```diff
@@ -0,0 +1,348 @@
+- `Ubuntu Help -- Repositories <https://help.ubuntu.com/community/Repositories>`_
+- `Ubuntu Help -- Repositories/Ubuntu <https://help.ubuntu.com/community/Repositories/Ubuntu>`_
+
+Landscape Repositories
```
```suggestion
Landscape repositories
```

s-makin commented on docs/explanation/archive.rst:
```diff
@@ -0,0 +1,348 @@
+Resources
+---------
+
+- `Ubuntu Release Cycle <https://ubuntu.com/about/release-cycle>`_
+- `Ubuntu Blog -- Ubuntu updates, releases and repositories explained <https://ubuntu.com/blog/ubuntu-updates-releases-and-repositories-explained>`_
+- `Ubuntu Server Docs -- Package Management <https://ubuntu.com/server/docs/package-management>`_
+- `Ubuntu Wiki -- Mirrors <https://wiki.ubuntu.com/Mirrors>`_
+
+- `Ubuntu Help -- Repositories <https://help.ubuntu.com/community/Repositories>`_
+- `Ubuntu Help -- Repositories/Ubuntu <https://help.ubuntu.com/community/Repositories/Ubuntu>`_
```
```suggestion
- `Ubuntu release cycle <https://ubuntu.com/about/release-cycle>`_
- `Ubuntu blog -- Ubuntu updates, releases and repositories explained <https://ubuntu.com/blog/ubuntu-updates-releases-and-repositories-explained>`_
- `Ubuntu Server docs -- package management <https://ubuntu.com/server/docs/package-management>`_
- `Ubuntu Wiki -- Mirrors <https://wiki.ubuntu.com/Mirrors>`_

- `Ubuntu Help -- Repositories <https://help.ubuntu.com/community/Repositories>`_
- `Ubuntu Help -- Repositories/Ubuntu <https://help.ubuntu.com/community/Repositories/Ubuntu>`_
```

s-makin commented on docs/explanation/archive.rst:
```diff
@@ -0,0 +1,348 @@
+Landscape Repositories
+~~~~~~~~~~~~~~~~~~~~~~
+
+Landscape is a management and administration tool for :term:`Ubuntu`.
+See more information `here <https://ubuntu.com/landscape>`_.
```
```suggestion
`Landscape <https://ubuntu.com/landscape>`_ is a management and administration tool for Ubuntu.
```

s-makin commented on docs/explanation/archive.rst:
```diff
@@ -0,0 +1,348 @@
+Landscape is a management and administration tool for :term:`Ubuntu`.
+See more information `here <https://ubuntu.com/landscape>`_.
+
+Landscape allows you to mirror :term:`APT` :ref:`Repositories` like the
+*Ubuntu Package Archive*. Although it is not directly related to the
+*Ubuntu Package Archive* it can be educational to understand how
+:term:`APT` :ref:`Repositories` work in general.
```
```suggestion
Landscape allows you to mirror APT repositories like the
Ubuntu package archive. Although it is not directly related to the
Ubuntu package archive, it can be educational to understand how
APT repositories work in general.
```

s-makin reviewed (commented):
Part 1/x - mostly nits, but some questions and suggestions :)

s-makin commented on docs/explanation/launchpad.rst:
```diff
@@ -0,0 +1,294 @@
+
+Synopsis
+^^^^^^^^
+
```
I got a little confused here, I'm not sure synopsis is very descriptive as a title. It might help to have a little sentence between the title and the code block to explain it a bit?

s-makin replied:
I'm also a bit confused by the table and what its purpose is - might be worth just clarifying that with a short introduction as well

dviererbe replied:
> I got a little confused here, I'm not sure synopsis is very descriptive as a title. It might help to have a little sentence between the title and the code block to explain it a bit?

I thought this is an established concept; like in this man page: https://manpages.ubuntu.com/manpages/lunar/en/man1/bash.1.html

But I agree that this can be confusing for someone not aware of this :/ There will be more sections like this where the synopsis will just show the syntax e.g. of a command. Do you have a general recommendation what to write there?

dviererbe replied:
> I'm also a bit confused by the table and what its purpose is - might be worth just clarifying that with a short introduction as well

Good Point! The input is what you type in a launchpad text field and the output shows how it will be rendered on Launchpad. I add a short description to explain this.

s-makin replied:
> > I got a little confused here, I'm not sure synopsis is very descriptive as a title. It might help to have a little sentence between the title and the code block to explain it a bit?
> 
> I thought this is an established concept; like in this man page: https://manpages.ubuntu.com/manpages/lunar/en/man1/bash.1.html
> 
> But I agree that this can be confusing for someone not aware of this :/ There will be more sections like this where the synopsis will just show the syntax e.g. of a command. Do you have a general recommendation what to write there?

If it's the expected terminology then stick with it, since the typical user of the packaging guide can be expected to know it. Let's chalk it up to "a thing I just needed to learn" :)

dviererbe replied:
I changed "Synopsis" to "Pattern" and moved the entire Launchpad comment section into a reference article as the section is barely explanatory content.

Resolved with 3d3ffc9a8b2db303ff33bbc35872d7d980f8ca3e

s-makin commented on docs/explanation/launchpad.rst:
```diff
@@ -0,0 +1,294 @@
+
+.. note::
+
+    This pattern is case invariant. The amount of :term:`Whitespace` can be 
+    variable, but if you place :term:`Whitespace` anywhere else; the 
```
```suggestion
    This pattern is case invariant. The amount of :term:`whitespace <Whitespace>` can be 
    variable, but if you place whitespace anywhere else the 
```

s-makin commented on docs/explanation/launchpad.rst:
```diff
@@ -0,0 +1,294 @@
+
+    This pattern is case invariant. The amount of :term:`Whitespace` can be 
+    variable, but if you place :term:`Whitespace` anywhere else; the 
+    :term:`Regular Expression` might not parse the input correctly. 
```
```suggestion
    :term:`regular expression <Regular Expression>` might not parse the input correctly. 
```

s-makin commented on docs/explanation/launchpad.rst:
```diff
@@ -0,0 +1,294 @@
+
+Launchpad will
+
+- cut off any :term:`Whitespace` to the right and
+- keep any :term:`Whitespace` to the left and
+- reduce any :term:`Whitespace` between non-:term:`Whitespace` characters to 
```
```suggestion
- cut off any whitespace to the right and
- keep any whitespace to the left and
- reduce any whitespace between non-whitespace characters to 
```

s-makin commented on docs/explanation/launchpad.rst:
```diff
@@ -0,0 +1,294 @@
+
+.. note::
+
+    Technically Launchpad passes :term:`Whitespace` through and 
+    the browser just ignores the :term:`Whitespace`.
```
```suggestion
    Technically, Launchpad passes whitespace through and 
    the browser just ignores the whitespace.
```

s-makin commented on docs/explanation/launchpad.rst:
```diff
@@ -0,0 +1,294 @@
+            | \| sit \| amet \| consectetur \|  
+            | \| adipiscing \| elit \| sed \|
+
+    or long chunks of :term:`Whitespace` between two sections:
```
```suggestion
    or long chunks of whitespace between two sections:
```

s-makin commented on docs/explanation/launchpad.rst:
```diff
@@ -0,0 +1,294 @@
+            | 
+            | But they're still just two paragraphs
+
+Getting Help
```
```suggestion
Getting help
```

s-makin commented on docs/explanation/launchpad.rst:
```diff
@@ -0,0 +1,294 @@
+
+If you need help with Launchpad you can choose any of the following methods:
+
+IRC Chatrooms
```
```suggestion
IRC chatrooms
```

s-makin commented on docs/explanation/launchpad.rst:
```diff
@@ -0,0 +1,294 @@
+IRC Chatrooms
+~~~~~~~~~~~~~
+
+On the ``irc.libera.chat`` :term:`IRC` server exists the ``#launchpad`` channel, where you
+can ask the Launchpad team and the :term:`Ubuntu` community for help.
```
```suggestion
On the ``irc.libera.chat`` :term:`IRC` server you will find the ``#launchpad`` channel, where you
can ask the Launchpad team and the Ubuntu community for help.
```

s-makin commented on docs/explanation/launchpad.rst:
```diff
@@ -0,0 +1,294 @@
+On the ``irc.libera.chat`` :term:`IRC` server exists the ``#launchpad`` channel, where you
+can ask the Launchpad team and the :term:`Ubuntu` community for help.
+
+Mailing Lists
```
```suggestion
Mailing lists
```

s-makin commented on docs/explanation/launchpad.rst:
```diff
@@ -0,0 +1,294 @@
+Mailing Lists
+~~~~~~~~~~~~~
+
+If you prefer to ask for help via E-Mail, you can write to the
```
```suggestion
If you prefer to ask for help via email, you can write to the
```

s-makin commented on docs/explanation/launchpad.rst:
```diff
@@ -0,0 +1,294 @@
+Ask a question
+~~~~~~~~~~~~~~
+
+As mentioned above, Launchpad has a community FAQ feature (called "Answers") where
+you can see other people's questions or ask one yourself. Use can use the *Answers*
+feature of the Launchpad project on Launchpad itself: https://answers.launchpad.net/launchpad
```
```suggestion
As mentioned above, Launchpad has a `community FAQ feature <https://answers.launchpad.net/launchpad>`_ called "Answers" where
you can see other people's questions or ask one yourself.
```

s-makin commented on docs/explanation/launchpad.rst:
```diff
@@ -0,0 +1,294 @@
+Report a bug
+~~~~~~~~~~~~
+
+If you encounter any bug related to Launchpad, you can submit a bug report to the
+:term:`Bug Tracking System` of the Launchpad project on Launchpad itself: 
+https://bugs.launchpad.net/launchpad
```
```suggestion
If you encounter any bug related to Launchpad, you can submit a bug report to the
:term:`Bug Tracking System` of the Launchpad project `on Launchpad itself <https://bugs.launchpad.net/launchpad>`_.
```

s-makin commented on docs/explanation/launchpad.rst:
```diff
@@ -0,0 +1,294 @@
+:term:`Bug Tracking System` of the Launchpad project on Launchpad itself: 
+https://bugs.launchpad.net/launchpad
+
+Staging Environment
```
```suggestion
Staging environment
```

s-makin commented on docs/explanation/launchpad.rst:
```diff
@@ -0,0 +1,294 @@
+-------------------
+
+Before new features are deployed to the production environment they get
+deployed to a staging environment (https://qastaging.launchpad.net/) where
```
```suggestion
`deployed to a staging environment <https://qastaging.launchpad.net/>`_ where
```

s-makin commented on docs/explanation/launchpad.rst:
```diff
@@ -0,0 +1,294 @@
+deployed to a staging environment (https://qastaging.launchpad.net/) where
+the changes can get tested.
+
+You can use the staging environment, to try out :term:`Launchpad` features.
```
```suggestion
You can use the staging environment, to try out Launchpad features.
```

s-makin commented on docs/explanation/launchpad.rst:
```diff
@@ -0,0 +1,294 @@
+---
+
+Launchpad has a Web :term:`API` that you can use to interact with its services.
+This makes it easy for developer communities like :term:`Ubuntu's <Ubuntu>` to
```
```suggestion
This makes it easy for developer communities like Ubuntu's to
```

s-makin commented on docs/explanation/launchpad.rst:
```diff
@@ -0,0 +1,294 @@
+This makes it easy for developer communities like :term:`Ubuntu's <Ubuntu>` to
+automate specific workflows.
+
+You can find the reference documentation for the Web :term:`API`:
+https://launchpad.net/+apidoc/
```
```suggestion
You can find the reference `documentation for the web API <https://launchpad.net/+apidoc/>`_ on Launchpad.
```

s-makin commented on docs/explanation/launchpad.rst:
```diff
@@ -0,0 +1,294 @@
+https://launchpad.net/+apidoc/
+
+The Launchpad team even created an :term:`Open Source <Open Source Software>`
+ython library ``launchpadlib``: https://help.launchpad.net/API/launchpadlib
```
```suggestion
Python library, `launchpadlib <https://help.launchpad.net/API/launchpadlib>`_
```

s-makin commented on docs/explanation/launchpad.rst:
```diff
@@ -0,0 +1,294 @@
+Resources
+---------
+
+- `Launchpad Home Page <Launchpad_>`_
+- `The Launchpad software project on Launchpad itself <https://launchpad.net/launchpad>`_
+    - `Launchpad Bug Tracker <https://bugs.launchpad.net/launchpad>`_
+    - `Launchpad Questions & Answers <https://answers.launchpad.net/launchpad>`_
+- `Launchpad Wiki <https://help.launchpad.net/>`_
+- `Launchpad Development Wiki <https://dev.launchpad.net/>`_
+- `Launchpad Blog <https://blog.launchpad.net/>`_
+- :doc:`/reference/git-ubuntu`
```
```suggestion
- `Launchpad home page <Launchpad_>`_
- `The Launchpad software project on Launchpad itself <https://launchpad.net/launchpad>`_
    - `Launchpad bug tracker <https://bugs.launchpad.net/launchpad>`_
    - `Launchpad questions and answers <https://answers.launchpad.net/launchpad>`_
- `Launchpad Wiki <https://help.launchpad.net/>`_
- `Launchpad Development Wiki <https://dev.launchpad.net/>`_
- `Launchpad blog <https://blog.launchpad.net/>`_
- :doc:`/reference/git-ubuntu`
```

s-makin commented on docs/explanation/main-inclusion-review.rst:
```diff
@@ -0,0 +1,74 @@
+
+    Do not confuse the abbreviation :term:`MIR` with the `display server <https://mir-server.io/>`_.
+
+:term:`Packages <Package>` in :term:`Main` and :term:`Restricted` are officially
```
```suggestion
:term:`Packages <Package>` in :term:`main <Main>` and :term:`restricted <Restricted>` are officially
```

s-makin commented on docs/explanation/main-inclusion-review.rst:
```diff
@@ -0,0 +1,74 @@
+
+:term:`Packages <Package>` in :term:`Main` and :term:`Restricted` are officially
+maintained, supported and recommended by the :term:`Ubuntu` project.
+:term:`Canonical's <Canonical>` support services applies to these :term:`Packages <Package>`, 
+which include security updates and certain :term:`SLA` grantees when bugs are reported
+and technical support is requested.
```
```suggestion
:term:`Canonical's <Canonical>` support services apply to these packages, 
which include security updates and certain :term:`SLA` guarantees when bugs are reported
and technical support is requested.
```

s-makin commented on docs/explanation/main-inclusion-review.rst:
```diff
@@ -0,0 +1,74 @@
+- from :term:`Universe` to :term:`Main`
+- or from :term:`Multiverse` to :term:`Restricted`.
+
+This review process is called *Main Inclusion Review (MIR)*.
```
```suggestion
This review process is called **Main Inclusion Review (MIR)**.
```

s-makin commented on docs/explanation/main-inclusion-review.rst:
```diff
@@ -0,0 +1,74 @@
+
+This review process is called *Main Inclusion Review (MIR)*.
+
+Apply a package for Main Inclusion Review
```
This is a slightly odd ordering of words, although I think I know what you meant. I'd suggest either "Main Inclusion Review package application" or "Submit a package for Main Inclusion Review". You could also have "Apply for Main Inclusion Review for a package", although that's slightly longer.

dviererbe replied:
Now that mention it I see it too.

dviererbe replied:
Resolved with 89bec8a88797da66ab13f3414a42b5e33e2670fd

s-makin commented on docs/explanation/main-inclusion-review.rst:
```diff
@@ -0,0 +1,74 @@
+.. note::
+
+    The guidelines and review process is constantly evolving. Therefore you should
+    re-read the `Main Inclusion Review Documentation <MainInclusionReviewDocumentation_>`_
+    even if you have applied a package for :term:`Main Inclusion Review` in the past.
```
```suggestion
    re-read the `Main Inclusion Review documentation <MainInclusionReviewDocumentation_>`_
    even if you have submitted a package for Main Inclusion Review in the past.
```

s-makin commented on docs/explanation/main-inclusion-review.rst:
```diff
@@ -0,0 +1,74 @@
+    re-read the `Main Inclusion Review Documentation <MainInclusionReviewDocumentation_>`_
+    even if you have applied a package for :term:`Main Inclusion Review` in the past.
+
+    The `Main Inclusion Review Documentation <MainInclusionReviewDocumentation_>`_ is
```
```suggestion
    The Main Inclusion Review documentation is
```

s-makin commented on docs/explanation/main-inclusion-review.rst:
```diff
@@ -0,0 +1,74 @@
+    is also a living document. External contributions, suggestions, discussions or
+    questions about the process are always welcome.
+
+MIR Team weekly Meeting
```
```suggestion
MIR team weekly meeting
```

s-makin commented on docs/explanation/main-inclusion-review.rst:
```diff
@@ -0,0 +1,74 @@
+You can follow these `instructions <https://libera.chat/guides/connect>`_ on
+how to connect to ``irc.libera.chat``.
+
+The purpose of the meeting is
```
```suggestion
The purpose of the meeting is:
```

s-makin commented on docs/explanation/main-inclusion-review.rst:
```diff
@@ -0,0 +1,74 @@
+
+Due to the nature of the :term:`Main Inclusion Review` process there are times
+when this meeting is very busy. Usually at the beginning of a new :term:`Ubuntu`
+release not a lot is happening.
```
This makes it sound like everyone's just chilling lol. I'd recommend rephrasing to something like "Usually, increased activity comes with the period just before a new Ubuntu release, and after that the meetings tend to be quieter." or something like that

dviererbe replied:
lol :D To be honest I have seen MIR meetings in the beginning of a cycle where nothing happened, because there was no MIR request yet.

I agree, I will also suggest changing that in the MIR documentation too ^^

dviererbe replied:
Resolved with 308f8c23277517bf6c5a7c2d901f0fbdbee36ed6

s-makin reviewed (commented):
Part 2/x

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+
+.. _NativeSourcePackages:
+
+(Non-)Native Source Packages
```
```suggestion
Native source packages
```

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In most cases, a software project is packaged by external contributors
+(:term:`Maintainers <Maintainer>`) only tangentially related to the software project. 
```
```suggestion
(:term:`maintainers <Maintainer>`) only tangentially related to the software project. 
```

s-makin replied:
In what sense do you mean "tangentially"?

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+
+In most cases, a software project is packaged by external contributors
+(:term:`Maintainers <Maintainer>`) only tangentially related to the software project. 
+Often, the *source package* has to do modifications to solve specific problems for its target
+:term:`Distribution`. The *source package* can, in these cases, be considered as its own
```
```suggestion
Often, the source package has to do modifications to solve specific problems for its target
:term:`distribution <Distribution>`. The source package can, in these cases, be considered as its own
```

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+:term:`Distribution`. The *source package* can, in these cases, be considered as its own
+software project, like a fork.
+
+Consequently, the :term:`Upstream` releases and *source package* releases do not always align.
```
```suggestion
Consequently, the :term:`Upstream` releases and source package releases do not always align.
```

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+Consequently, the :term:`Upstream` releases and *source package* releases do not always align.
+
+Native packages almost always originate from software projects designed with :term:`Debian`
+packaging in mind and have no independent existence outside its target :term:`Distribution`.
```
I'm not sure whether "its" in this case refers to native packages, software projects, or Debian software projects specifically

dviererbe replied:
I changed the wording with d95ce7b43232c1ae4f27b55c427d3369f63465e8

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+Consequently, the :term:`Upstream` releases and *source package* releases do not always align.
+
+Native packages almost always originate from software projects designed with :term:`Debian`
+packaging in mind and have no independent existence outside its target :term:`Distribution`.
```
```suggestion
packaging in mind and have no independent existence outside its target :term:`distribution <Distribution>`.
```

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+
+Native packages almost always originate from software projects designed with :term:`Debian`
+packaging in mind and have no independent existence outside its target :term:`Distribution`.
+Native packages do not differentiate between :term:`Upstream` releases and *source package* releases.
+Therefore, the version identifier of a native package does not have a 
+:term:`Ubuntu` or :term:`Debian` specific component.
```
```suggestion
Native packages do not differentiate between Upstream releases and source package releases.
Therefore, the version identifier of a native package does not have an 
Ubuntu- or Debian-specific component.
```

s-makin replied:
I'm not 100% sure I could define, after reading this, what a native package actually *is*.

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+Format: ``3.0 (quilt)``
+^^^^^^^^^^^^^^^^^^^^^^^
+
+A new-generation *source package* format that records modifications in a :manpage:`quilt(1)`
+:term:`Patch` series within the :file:`debian/patches` folder. The :term:`Patches <Patch>` are
+organized as a :term:`Stack`, and you can apply, un-apply, and update them easily by traversing 
+the :term:`Stack` (push/pop). These changes are automatically applied during the extraction of
+the *source package*.
```
```suggestion
A new-generation source package format that records modifications in a :manpage:`quilt(1)`
:term:`patch <Patch>` series within the :file:`debian/patches` folder. The patches are
organized as a :term:`stack <Stack>`, and you can apply, un-apply, and update them easily by traversing 
the stack (push/pop). These changes are automatically applied during the extraction of
the source package.
```

s-makin replied:
push/pop is a bit confusing without any context. I'd recommend either removing it, or adding the missing context.

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+the :term:`Stack` (push/pop). These changes are automatically applied during the extraction of
+the *source package*.
+
+A *source package* in this format contains at least an original tarball (``.orig.tar.ext`` where
```
```suggestion
A source package in this format contains at least an original tarball (``.orig.tar.ext`` where
```

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+the *source package*.
+
+A *source package* in this format contains at least an original tarball (``.orig.tar.ext`` where
+``ext`` can be ``gz``, ``bz2``, ``lzma`` and ``xz``) and a debian tarball (``.debian.tar.ext``).
```
```suggestion
``ext`` can be ``gz``, ``bz2``, ``lzma`` or ``xz``) and a Debian tarball (``.debian.tar.ext``).
```

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+``ext`` can be ``gz``, ``bz2``, ``lzma`` and ``xz``) and a debian tarball (``.debian.tar.ext``).
+It can also contain additional original tarballs (``.orig-component.tar.ext``), where ``component``
+can only contain alphanumeric (``a-z``, ``A-Z``, ``0-9``) characters and hyphens (``-``).
+Optionally, each original tarball can be accompanied by a detached upstream signature
```
```suggestion
Optionally, each original tarball can be accompanied by a detached Upstream signature
```

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+
+.. code:: bash
+
+    pull-lp-source --download-only 'hello' '2.10-3'
```
This command doesn't work without first installing ubuntu-dev-tools

s-makin replied:
After I installed, I got the following output:

Found hello 2.10-3 in mantic
Downloading hello_2.10-3.dsc from archive.ubuntu.com (0.002 MiB)
[=====================================================>]100%
Good signature by Santiago Vila <sanvila@debian.org> (0x41CE7F0B9F1B8B32)
Downloading hello_2.10.orig.tar.gz from archive.ubuntu.com (0.692 MiB)
[=====================================================>]100%
Downloading hello_2.10.orig.tar.gz.asc from archive.ubuntu.com (0.001 MiB)
[=====================================================>]100%
Downloading hello_2.10-3.debian.tar.xz from archive.ubuntu.com (0.012 MiB)
[=====================================================>]100%

dviererbe replied:
Installing `ubuntu-dev-tools` will be part of the getting set up tutorial. I add a note.

dviererbe replied:
I added a note to the first pull-lp-source to install ubuntu-dev-tools

Resolved with d95ce7b43232c1ae4f27b55c427d3369f63465e8

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+
+    pull-lp-source --download-only 'hello' '2.10-3'
+
+Now you should see the following files:
```
By "see" do you just mean the download summary, or is there some other command you'd need to run to just view the files as they are listed below?

s-makin replied:
Ah, I see them when I do `ls`

dviererbe replied:
Yes, my intention was to run `ls` afterwards, but theoretically both is fine. The download summary also shows the downloaded list of files.

dviererbe replied:
I now encourage to run ls.

Resolved with d95ce7b43232c1ae4f27b55c427d3369f63465e8

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+
+Now you should see the following files:
+
+- :file:`hello_2.10-3.dsc`: The *Debian Source Control* file of the *source package*.
```
```suggestion
- :file:`hello_2.10-3.dsc`: The Debian Source Control file of the source package.
```

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+Now you should see the following files:
+
+- :file:`hello_2.10-3.dsc`: The *Debian Source Control* file of the *source package*.
+- :file:`hello_2.10.orig.tar.gz`: The tarball containing the original :term:`Source Code` of the :term:`Upstream` project.
```
```suggestion
- :file:`hello_2.10.orig.tar.gz`: The tarball containing the original source code of the Upstream project.
```

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+
+- :file:`hello_2.10-3.dsc`: The *Debian Source Control* file of the *source package*.
+- :file:`hello_2.10.orig.tar.gz`: The tarball containing the original :term:`Source Code` of the :term:`Upstream` project.
+- :file:`hello_2.10.orig.tar.gz.asc`: The detached :term:`Upstream` signature of :file:`hello_2.10.orig.tar.gz`.
```
```suggestion
- :file:`hello_2.10.orig.tar.gz.asc`: The detached Upstream signature of :file:`hello_2.10.orig.tar.gz`.
```

s-makin replied:
I'm not sure what a Detached upstream signature is

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+- :file:`hello_2.10-3.dsc`: The *Debian Source Control* file of the *source package*.
+- :file:`hello_2.10.orig.tar.gz`: The tarball containing the original :term:`Source Code` of the :term:`Upstream` project.
+- :file:`hello_2.10.orig.tar.gz.asc`: The detached :term:`Upstream` signature of :file:`hello_2.10.orig.tar.gz`.
+- :file:`hello_2.10-3.debian.tar.xz`: The tarball containing the content of the debian directory.
```
```suggestion
- :file:`hello_2.10-3.debian.tar.xz`: The tarball containing the content of the Debian directory.
```

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+Format: ``3.0 (native)``
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+A new-generation *source package* format extends the native package
+format defined in the :ref:`format 1.0 <SourcePackageFormat_1.0>`.
+
+A *source package* in this format is a tarball (``.tar.ext`` where ``ext``
+can be ``gz``, ``bz2``, ``lzma`` and ``xz``).
```
```suggestion
A new-generation source package format that extends the native package
format defined in the :ref:`format 1.0 <SourcePackageFormat_1.0>`. A source
package in this format is a tarball (``.tar.ext`` where ``ext``
can be ``gz``, ``bz2``, ``lzma`` or ``xz``).
```

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+A *source package* in this format is a tarball (``.tar.ext`` where ``ext``
+can be ``gz``, ``bz2``, ``lzma`` and ``xz``).
+
+For example, take a look at the ``subiquity`` package:
```
```suggestion
For example, let's take a look at the ``subiquity`` package:
```

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+
+Now you should see the following files:
+
+- ``ubiquity_23.10.2.dsc``:  The *Debian Source Control* file of the *source package*.
+- ``ubiquity_23.10.2.tar.xz``: The tarball containing the :term:`Source Code` of the project.
```
```suggestion
- ``ubiquity_23.10.2.dsc``:  The Debian Source Control file of the source package.
- ``ubiquity_23.10.2.tar.xz``: The tarball containing the source code of the project.
```

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+Format: ``1.0``
+^^^^^^^^^^^^^^^
+
+The original *source package* format. Nowadays, this format is rarely used.
+
+A native *source package* in this format consists of a single ``.tar.gz``
+file containing the :term:`Source`.
```
```suggestion
The original source package format. Nowadays, this format is rarely used.
A native source package in this format consists of a single ``.tar.gz``
file containing the source.
```

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+A native *source package* in this format consists of a single ``.tar.gz``
+file containing the :term:`Source`.
+
+A non-native *source package* in this format consists of a ``.orig.tar.gz`` file
+(containing the :term:`Upstream` :term:`Source`) associated with a ``.diff.gz``
+file (the :term:`Patch` containing :term:`Debian` packaging modifications).
+Optionally, the original tarball can be accompanied by a detached :term:`Upstream`
+signature ``.orig.tar.gz.asc``. 
```
```suggestion
A non-native source package in this format consists of an ``.orig.tar.gz`` file
(containing the Upstream source) associated with a ``.diff.gz``
file (the patch containing Debian packaging modifications).
Optionally, the original tarball can be accompanied by a detached Upstream
signature ``.orig.tar.gz.asc``. 
```

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+
+.. note::
+  
+  This format does not specify a :term:`Patch` system, which makes it harder for
+  :term:`Maintainers <Maintainer>` to track modifications. There were multiple approaches
+  to how packages tracked :term:`Patches <Patch>`. Therefore, the *source packages* of this
+  format often contained a :file:`debian/README.source` file explaining how to use the 
+  :term:`Patch` system.
```
```suggestion
  This format does not specify a patch system, which makes it harder for
  maintainers to track modifications. There were multiple approaches
  to how packages tracked patches. Therefore, the source packages of this
  format often contained a :file:`debian/README.source` file explaining how to use the 
  patch system.
```

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+  format often contained a :file:`debian/README.source` file explaining how to use the 
+  :term:`Patch` system.
+
+``3.0`` Formats Improvements
```
```suggestion
``3.0`` Format improvements
```

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+Some of the improvements that apply to most ``3.0`` formats are:
+
+- Support for additional compression formats: ``bzip2``, ``lzma`` and ``xz``
+- Support for multiple :term:`Upstream` tarballs
```
```suggestion
- Support for multiple Upstream tarballs.
```

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+
+Some of the improvements that apply to most ``3.0`` formats are:
+
+- Support for additional compression formats: ``bzip2``, ``lzma`` and ``xz``
```
```suggestion
- Support for additional compression formats: ``bzip2``, ``lzma`` and ``xz``.
```

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+
+- Support for additional compression formats: ``bzip2``, ``lzma`` and ``xz``
+- Support for multiple :term:`Upstream` tarballs
+- Supports inclusion of binary files
```
```suggestion
- Support for inclusion of binary files.
```

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+- Support for additional compression formats: ``bzip2``, ``lzma`` and ``xz``
+- Support for multiple :term:`Upstream` tarballs
+- Supports inclusion of binary files
+- :term:`Debian`-specific changes are no longer stored in a single ``.diff.gz``
```
```suggestion
- Debian-specific changes are no longer stored in a single ``.diff.gz``.
```

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+- Support for multiple :term:`Upstream` tarballs
+- Supports inclusion of binary files
+- :term:`Debian`-specific changes are no longer stored in a single ``.diff.gz``
+- The :term:`Upstream` tarball does not need to be repacked to strip the :term:`Debian` directory.
```
```suggestion
- The Upstream tarball does not need to be repacked to strip the Debian directory.
```

s-makin replied:
I don't know what "strip" means in this context

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+- :term:`Debian`-specific changes are no longer stored in a single ``.diff.gz``
+- The :term:`Upstream` tarball does not need to be repacked to strip the :term:`Debian` directory.
+
+Other Formats
```
```suggestion
Other formats
```

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+^^^^^^^^^^^^^
+
+The following formats are rarely used, experimental and/or historical.
+You should only choose these if you know what you do.
```
```suggestion
You should only choose these if you know what you are doing.
```

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+The following formats are rarely used, experimental and/or historical.
+You should only choose these if you know what you do.
+
+- ``3.0 (custom)``: Doesn't represent an actual *source package* format but can
+  be used to create *source packages* with arbitrary files.
+- ``3.0 (git)``: An experimental format to package from a :term:`git` repository.
+- ``3.0 (bzr)``: An experimental format to package from a :term:`Bazaar` repository.
+- ``2.0``: The first specification of a new-generation *source package* format.
+  It was never widely adopted and eventually replaced by
+  :ref:`3.0 (quilt) <SourcePackageFormat_3.0quilt>`.
```
```suggestion
- ``3.0 (custom)``: Doesn't represent an actual source package format but can
  be used to create source packages with arbitrary files.
- ``3.0 (git)``: An experimental format to package from a :term:`git` repository.
- ``3.0 (bzr)``: An experimental format to package from a :term:`Bazaar` repository.
- ``2.0``: The first specification of a new-generation source package format.
  It was never widely adopted and eventually replaced by
  :ref:`3.0 (quilt) <SourcePackageFormat_3.0quilt>`.
```

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+``.changes`` file
+~~~~~~~~~~~~~~~~~
+
+Although technically not part of a *source package* -- every time a *source package* is built,
+a :file:`.changes` file will be created alongside. The :file:`.changes` file contains metadata
+from the *Source Control* file and other information (e.g. the latest changelog entry) about the
+*source package*. :term:`Archive` tools and :term:`Archive Administrators <Archive Admin>` use this
+data to process changes to *source packages* and determine the appropriate action to upload the 
+*source package* to the :term:`Ubuntu Archive`.
```
```suggestion
Although technically not part of a source package -- every time a source package is built,
a :file:`.changes` file will be created alongside it. The :file:`.changes` file contains metadata
from the Source Control file and other information (e.g. the latest changelog entry) about the
source package. :term:`Archive` tools and :term:`Archive Administrators <Archive Admin>` use this
data to process changes to source packages and determine the appropriate action to upload the 
source package to the :term:`Ubuntu Archive`.
```

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+
+.. _BinaryPackages:
+
+Binary Packages
```
```suggestion
Binary packages
```

s-makin replied:
I would actually recommend moving this to above the source packages section, since this one is so short and binary packages are referenced elsewhere - I think it's important not to bury this bit at the end of the page

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+Binary Packages
+---------------
+
+A *binary package* is a standardized format that the :term:`Package Manager` (:manpage:`dpkg(1)` or
```
```suggestion
A **binary package** is a standardized format that the :term:`Package Manager` (:manpage:`dpkg(1)` or
```

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+---------------
+
+A *binary package* is a standardized format that the :term:`Package Manager` (:manpage:`dpkg(1)` or
+:manpage:`apt(8)`) can understand to install and uninstall software on a target machine to simplify
+distributing software to a target machine and managing software on a target machine.
```
```suggestion
:manpage:`apt(8)`) can understand to install and uninstall software on a target machine. This simplifies
distributing software to a target machine and managing the software on that machine.
```

s-makin commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+:manpage:`apt(8)`) can understand to install and uninstall software on a target machine to simplify
+distributing software to a target machine and managing software on a target machine.
+
+A :term:`Debian` *binary package* is a file with the :file:`.deb` file extension that
```
```suggestion
A Debian binary package uses the :file:`.deb` file extension and
```

s-makin commented on docs/explanation/proposed-migrations.rst:
```diff
@@ -1,9 +1,2 @@
-.. _proposed-migrations:
-
-===================
 Proposed Migrations
```
```suggestion
Proposed migrations
```

s-makin reviewed (commented):
3/x

s-makin commented on docs/explanation/releases.rst:
```diff
@@ -0,0 +1,229 @@
+~~~~~~~~~~~~~~~~
+
+Every other release, between :ref:`LTSReleases`, is an *"interim release"* that is
+also often called a *"regular release"*.
```
Saying "every other release" has the potential to be read a bit ambiguously. I'd recommend rephrasing as something like:
"In the years between LTS releases, Canonical also produces **interim releases*, sometimes also called a "regular release".

dviererbe replied:
Agree, fixed with d95ce7b43232c1ae4f27b55c427d3369f63465e8

s-makin commented on docs/explanation/releases.rst:
```diff
@@ -0,0 +1,229 @@
+
+Every :term:`Ubuntu` :term:`Series` receives the same production-grade support quality,
+but the length of the period in which an :term:`Ubuntu` :term:`Series` receives support
+differs drastically.
```
I'd say "varies" rather than "differs drastically" (which sounds quite dramatic :sweat_smile:)

dviererbe replied:
Agree; I see you already fixed this yourself.

s-makin commented on docs/explanation/releases.rst:
```diff
@@ -0,0 +1,229 @@
+Every :term:`Ubuntu` release is provided as a :term:`Server <Ubuntu Server>` and
+:term:`Desktop <Ubuntu Desktop>` edition.
+
+:term:`Ubuntu Desktop` is designed for end-users and provides a graphical :term:`User Interface`
```
I'm not sure about the use of "end-users" here. Server is also designed for end-users, and it has a variety of use-cases. I'd focus on the main difference (which you already highlighted, being the GUI vs TUI) by just removing that bit:

"Ubuntu Desktop provides a graphical user interface for..."

dviererbe replied:
Agree, resolved with 219db6c85b1ecccd69d0f2a26f49d3fcad367650

s-makin commented on docs/explanation/releases.rst:
```diff
@@ -0,0 +1,229 @@
+:term:`Ubuntu Desktop` is designed for end-users and provides a graphical :term:`User Interface`
+(:term:`GUI`) for everyday computing tasks, making it suitable for personal computers
+and laptops. :term:`Ubuntu Server`, on the other hand, is a command-line-based
+:term:`Operating System` optimized for server environments, lacking a :term:`GUI`
```
"lacking a GUI" makes it sound like an oversight that we haven't implemented one - in reality we have a TUI instead of a GUI that allows machines on the server to be run headless.

dviererbe replied:
Agree, resolved with 1f035bbe9c6fbbc3c0087e54a3d5d5ebd827f181

s-makin commented on docs/explanation/releases.rst:
```diff
@@ -0,0 +1,229 @@
+
+:term:`Canonical` publishes :term:`Ubuntu` on all major public clouds, and the latest
+:term:`image` for each :ref:`LTS <LTSReleases>` version will always include security updates
+rolled up to at most two weeks ago.
```
"rolled up" sounds quite colloquial and may not be easily understood. Also "two weeks ago" sounds like it depends on the date of writing and not the date of download. I'd suggest something more like "...include any security updates provided since the LTS release date, until two weeks prior to the image creation date."
(this is also a bit ugly so maybe tweak to your satisfaction)

dviererbe replied:
Agree, resolved with fb1c2e81ea7de3549f0eee932209d57e889f3fbd

s-makin commented on docs/explanation/upstream-and-downstream.rst:
```diff
@@ -1,9 +1,139 @@
+carries modifications, improvements, and code is used. It describes the relationship and direction
+of changes made between projects. This stream originates (upwards) from the original project
+(and related entities like :term:`Source Code`, authors, and :term:`maintainers <Maintainer>`) and
+flows downwards to projects (and associated entities) that depend on it.
```
This feels like a bit like a preamble, so I'd suggest putting it between the Terminology heading and the delta subheading

dviererbe replied:
Sounds good to me. It was originally there, before I pushed it, but I thought it is weird that I first explain this analogy and then introduce "Ubuntu delta".

dviererbe replied:
I put this to resolved as you already moved it yourself.

s-makin commented on docs/explanation/upstream-and-downstream.rst:
```diff
@@ -1,9 +1,139 @@
+
+Ubuntu delta (noun):
+    A modification to an :term:`Ubuntu` :term:`Package` that is derived from a :term:`Debian`
+    :term:`Package`.
```
From what I understand (I fully expect I'm wrong), the delta is not so much a modification to an Ubuntu package, but rather the difference (in the mathematical sense) between the Debian package and the Ubuntu one - which might represent one modification, or several.

dviererbe replied:
That is totally correct :) Can I change something to make a reader more confident after reading the Ubuntu delta explanation?

s-makin commented on docs/explanation/upstream-and-downstream.rst:
```diff
@@ -1,9 +1,139 @@
+
+    *Examples*
+        - :term:`Ubuntu` is a downstream project of :term:`Debian`.
+        - *"The bug is already patched downstream."*
+        - *"The bug was reported by a downstream user."*
+        - *"Downstream maintainers have submitted a bugfix."*
+        - *"The change may affect downstream users."*
```
```suggestion
        - The bug is already patched downstream.
        - The bug was reported by a downstream user.
        - Downstream maintainers have submitted a bugfix.
        - The change may affect downstream users.
```

s-makin commented on docs/explanation/upstream-and-downstream.rst:
```diff
@@ -1,9 +1,139 @@
+        - *"Downstream maintainers have submitted a bugfix."*
+        - *"The change may affect downstream users."*
+
+*downstream (verb)*
```
```suggestion
Downstream (verb):
```

s-makin commented on docs/explanation/upstream-and-downstream.rst:
```diff
@@ -1,9 +1,139 @@
+    Similar to :ref:`upstream_verb`: Sending something (usually a :term:`Patch`) downstream
+    that originated from an upstream project.
+
+    *Example*
```
```suggestion
    *Example*:
```

s-makin commented on docs/explanation/upstream-and-downstream.rst:
```diff
@@ -1,9 +1,139 @@
+    that originated from an upstream project.
+
+    *Example*
+        - "We downstreamed the :term:`Patch`."
```
```suggestion
        - We downstreamed the :term:`Patch`.
```

s-makin commented on docs/explanation/upstream-and-downstream.rst:
```diff
@@ -1,9 +1,139 @@
+    *Example*
+        - "We downstreamed the :term:`Patch`."
+
+Why upstream changes?
```
```suggestion
Why do we upstream changes?
```

s-makin commented on docs/explanation/upstream-and-downstream.rst:
```diff
@@ -1,9 +1,139 @@
+---------------------
+
+.. note::
+    The following list does not aim for completeness. There are certainly more good arguments 
```
```suggestion
    The following list does not aim for completeness. There are plenty more good arguments 
```

s-makin replied:
("certainly more good" could be read like there are better arguments that aren't listed)

s-makin commented on docs/explanation/upstream-and-downstream.rst:
```diff
@@ -1,9 +1,139 @@
 
-    Work in progress!
\ No newline at end of file
+- **Decreased Maintenance Complexity**: Think of any :term:`Ubuntu` :term:`Package` derived from a 
```
```suggestion
- **Decreased maintenance complexity**: Think of any :term:`Ubuntu` :term:`Package` derived from a 
```

s-makin commented on docs/explanation/upstream-and-downstream.rst:
```diff
@@ -1,9 +1,139 @@
+  :term:`Debian` :term:`Package` that carries a :term:`delta <Ubuntu Delta>`. Every time the
+  :term:`Debian` :term:`Package` gets updated, the :term:`Ubuntu` :term:`Package` may be subject
+  to a :term:`Merge Conflict` when the changes to the :term:`Debian` :term:`Package` get applied to
+  the :term:`Ubuntu` :term:`Package`. Upstreaming changes reduces the maintenance cost to resolve
```
```suggestion
  the :term:`Ubuntu` :term:`Package`. By upstreaming changes we reduce the maintenance cost to resolve
```

s-makin commented on docs/explanation/upstream-and-downstream.rst:
```diff
@@ -1,9 +1,139 @@
+  to a :term:`Merge Conflict` when the changes to the :term:`Debian` :term:`Package` get applied to
+  the :term:`Ubuntu` :term:`Package`. Upstreaming changes reduces the maintenance cost to resolve
+  :term:`Merge Conflicts <Merge Conflict>` when they occur.
+- **Quality Assurance & Security**: Any changes that get upstreamed will also be subject to the
```
```suggestion
- **Quality assurance and security**: Any changes that get upstreamed will also be subject to the
```

s-makin commented on docs/explanation/upstream-and-downstream.rst:
```diff
@@ -1,9 +1,139 @@
+  behaviour (especially security-related bugs). Also, be aware that an unpatched
+  :term:`security vulnerability <Common Vulnerabilities and Exposures>` in any system could lead to
+  the indirect exposure of other systems.
+- **Common Benefit**: By syncing the :term:`Debian` :term:`packages <Package>` into the
```
```suggestion
- **Common benefit**: By syncing the :term:`Debian` :term:`packages <Package>` into the
```

s-makin reviewed (commented):
Lost track of what number I'm on, but from this review and onward I've stopped making suggestions for the defined terms. I suggest that a global search is applied for terms in these pages and that the definition be removed from all except the first ones on each page. I'd also recommend checking each page against the rendered version so you can see why I'm suggesting we remove the excess ones - having the blue links everywhere on the page has the effect of confusing the reader as they camouflage genuine links you might want the reader to pay attention to. In addition to that, using this much emphasis throughout has the effect of diluting the value the emphasis might otherwise bring. There might be occasions where defining more than once per page is justified - hence a global search rather than a global search-and-replace - but this will definitely be quicker for you than for me picking them up individually! :)

s-makin committed:
Apply suggestions from review 1

s-makin committed:
Apply more review feedback

s-makin committed:
Applying further suggestions

dviererbe commented on docs/explanation/archive.rst:
```diff
@@ -0,0 +1,348 @@
+    .. tab-item:: Tunisia (TN)
+
+        Tunisia does not have any third-party mirrors in its country. Therefore the
+        Tunisia *Country Mirror* is just the main *Ubuntu Package Archive* server
```
Good catch!

dviererbe replied:
Resolved with 735ff5c17bc525e16db7cbbb1dbe5fd0d884596a

s-makin commented on docs/explanation/launchpad.rst:
```diff
@@ -0,0 +1,292 @@
+IRC chatrooms
+~~~~~~~~~~~~~
+
+On the ``irc.libera.chat`` :term:`IRC` server you will find the ``#launchpad`` channel, where you
```
It would be helpful for readers if we actually linked directly to this channel on Libera

dviererbe replied:
Do you mean like this?

```rst
`#launchpad <irc://irc.libera.chat/launchpad>`_
```

Or do you mean to move it to the start of the article?

s-makin replied:
I mean exactly like that! :) So they can just one-click to go to the channel

s-makin committed:
Final polishing from review 1

s-makin commented on docs/how-to/get-package-source.rst:
```diff
@@ -1,194 +1,241 @@
-    $ git-ubuntu export-orig
 
-Example::
+To generate the orig tarballs for a given source package:
```
Should we have a glossary entry for "orig tarballs"?

dviererbe replied:
There actually is already an glossary entry ^^

I added the term ref; resolved with 3e0791b1877525cec82c0e675a5df2d955bd5d1a

s-makin commented on docs/how-to/get-package-source.rst:
```diff
@@ -1,194 +1,241 @@
 
-- `git ubuntu clone <GitUbuntuBlogArticle1_>`_
-- `Git Ubuntu: More on the imported repositories <GitUbuntuBlogArticle2_>`_
+- `git ubuntu clone <https://ubuntu.com/blog/git-ubuntu-clone>`_
+- `Git Ubuntu: More on the imported repositories <https://ubuntu.com/blog/git-ubuntu-more-on-the-imported-repositories>`_
```
I'm a bit reluctant to direct people to 6-year-old blog posts which are only going to fall further and further out of date. Do you know of anything more up to date that we can point to (in the absence of docs, which I know aren't ready yet).

I think we should at least move these down the bottom of the page in a "Further reading" section if we really want to include them. For now, I'd also suggest putting the git-ubuntu section at the bottom, otherwise we're essentially prioritising it even though it's not "The Way" yet. Once it becomes The Way, we can always put it back at the top, or give it its own page.

dviererbe replied:
> I'd also suggest putting the git-ubuntu section at the bottom, otherwise we're essentially prioritising it even though it's not "The Way" yet. Once it becomes The Way, we can always put it back at the top, or give it its own page.

From my experience git-ubuntu clone is totally fine for day to day use although it is officially not production ready yet. I would argue that we did our due diligence by having the warning at the start of the section.

To be totally honest, although git-ubuntu is not all the way there yet, I personally want to push it for adoption to new readers. I want to avoid that someone learns the old way.

dviererbe replied:
> I'm a bit reluctant to direct people to 6-year-old blog posts which are only going to fall further and further out of date. Do you know of anything more up to date that we can point to (in the absence of docs, which I know aren't ready yet).

Unfortunately this is the most recent stuff I could find :/ I totally share your dissatisfaction.

s-makin replied:
I'm not sure really what the solution is here. I guess we can leave these as they are, but I'll make a note for myself to talk to Robie about getting some sort of blog post or more up-to-date article about git-ubuntu out there.

s-makin commented on docs/how-to/get-package-source.rst:
```diff
@@ -1,194 +1,241 @@
-The ``dget`` command is part of the ``devscripts`` package. If you call it with 
-the URL of a ``.dsc`` or ``.changes`` file it acts as a source package aware ``wget``.
+The :command:`dget` command is part of the ``devscripts`` package. If you call
+it with the URL of a ``.dsc`` or ``.changes`` file it acts as a source package
+aware :manpage:`wget(1)`.
```
I found this wording "source package aware wget" to be very confusing and I'm not sure what you meant to say here.

dviererbe replied:
Agree, I expanded the explanation.

Resolved with fd05113bb2ab1fc82184e4ef5dba4ce4b3afaddc

s-makin reviewed (changes requested):
So I've finished my (round 1) review, and done a fair bit of nitpick polishing myself to save you the time and hassle. I also did some minor changes to styling for consistency (header levels etc) on pages where there were differences within the page. We might have to review again once the style guide is done, or we can make the style guide inclusive of what you've already started us off with here. I've left some comments/questions/suggestions that you'll still need to address. 

Overall, really nice work! No structural issues that I could see, everything was in a sensible order, which I think will be helpful to readers. And you have a real talent for explaining things clearly and simply. Well done :D

dviererbe commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In most cases, a software project is packaged by external contributors
+(:term:`Maintainers <Maintainer>`) only tangentially related to the software project. 
```
The maintainer/packager of a package are often 3rd parties from the perspective of a software project.

A good example is the Ubuntu tor browser. I don't know its current state, but the tor browser was once removed from the archive, because the tor project requested it. They had no control over the package version in the Ubuntu archive and were annoyed by users reporting bugs that were already fixed, because the users assumed that the package was reasonably up-to-date.

dviererbe replied:
I changed the wording with d95ce7b43232c1ae4f27b55c427d3369f63465e8

dviererbe committed:
chore(explanation/main-inclusion-review): change wording

as suggested by Sally:
> This makes it sound like everyone's just chilling lol. I'd recommend
> rephrasing to something like "Usually, increased activity comes with
> the period just before a new Ubuntu release, and after that the
> meetings tend to be quieter." or something like that

Co-authored-by: Sally Makin <sally.makin@canonical.com>

dviererbe committed:
fix(how-to/install-built-packages): clarify technical detail

I just noticed that `add-apt-repository` does not add any entry
to `/etc/apt/sources.list`.

dviererbe committed:
feat(explanation/archive, reference/glossary): Add definition for term "Ubuntu Base Packages"

dviererbe committed:
fix(explanation/launchpad): improve wording

main [...] archive server --> primary [...] archive server

Co-authored-by: Sally Makin <sally.makin@canonical.com>

dviererbe committed:
feat(explanation/archive): Add section Security update propagation

dviererbe committed:
fix(main-inclusion-review): minor rewording

Addresses feedback from Sally:
> This is a slightly odd ordering of words, although I think I know
> what you meant. I'd suggest either "Main Inclusion Review package
> application" or "Submit a package for Main Inclusion Review".
> You could also have "Apply for Main Inclusion Review for a package",
> although that's slightly longer.

Co-authored-by: Sally Makin <sally.makin@canonical.com>

dviererbe committed:
feat(explanation/launchpad): Seperate markup reference into its own article

This commit also adds examples and improves to the formatting of the
content that is moved.

dviererbe committed:
feat(reference/glossary): Add term 'Signature' and 'Detached Signature'

This commit also adds a referencess to the term 'Detached Signature'
in the explanation/package-model article.

This addresses some of the feedback by Sally Makin <sally.makin@canonical.com>.

dviererbe committed:
fix(glossary): small grammar mistake

dviererbe committed:
feat(glossary): Add term 'Text Encoding'

dviererbe committed:
feat(glossary): Add term & definition 'Namespace'

Note: Referenced in explanation/archive

dviererbe committed:
chore(explanation/archive): Move term reference

dviererbe committed:
feat(glossary): Add terms 'Discourse', 'Ubuntu Discourse', Canonical Discourse

dviererbe commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+packaging in mind and have no independent existence outside its target :term:`Distribution`.
+Native packages do not differentiate between :term:`Upstream` releases and *source package* releases.
+Therefore, the version identifier of a native package does not have a 
+:term:`Ubuntu` or :term:`Debian` specific component.
```
I can improve it a bit, but I think you have to see the native package examples "3.0 (native)" below to fully grasp it.

dviererbe replied:
I expanded my explanation of native packages even further.

Resolved with d95ce7b43232c1ae4f27b55c427d3369f63465e8

dviererbe committed:
feat(glossary): Add definition for term 'Stack'

dviererbe commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+:term:`Patch` series within the :file:`debian/patches` folder. The :term:`Patches <Patch>` are
+organized as a :term:`Stack`, and you can apply, un-apply, and update them easily by traversing 
+the :term:`Stack` (push/pop). These changes are automatically applied during the extraction of
+the *source package*.
```
I added a definition in the glossary that explains what a Stack is.

Resolved with af788aa78c4d773eb1f64f5d1088f580ea9d71f4

dviererbe committed:
fix(reference/launchpad-text-markup): correct image alt-text

dviererbe committed:
fix: reference toc-tree link

dviererbe committed:
chore(explanation/package-model): Improve native packages explanation

Addresses feedback from Sally:
> I'm not 100% sure I could define, after reading this, what a native
> package actually is.

> This command [pull-lp-source] doesn't work without first installing
> ubuntu-dev-tools

> By "see" do you just mean the download summary, or is there some
> other command you'd need to run to just view the files as they are
> listed below?

dviererbe commented on docs/explanation/package-model.rst:
```diff
@@ -1,9 +1,196 @@
+
+- :file:`hello_2.10-3.dsc`: The *Debian Source Control* file of the *source package*.
+- :file:`hello_2.10.orig.tar.gz`: The tarball containing the original :term:`Source Code` of the :term:`Upstream` project.
+- :file:`hello_2.10.orig.tar.gz.asc`: The detached :term:`Upstream` signature of :file:`hello_2.10.orig.tar.gz`.
```
I added an explanation to the glossary with 6d99a955667ed9dd14110f33d3a526a71a1bd871

dviererbe committed:
chore(explanation/releases): change ambiguously wording

dviererbe committed:
chore(explanation/releases): changed questionable wording

dviererbe committed:
chore(explanation/releases): changed questionable wording

dviererbe committed:
chore(explanation/releases): changed questionable wording

dviererbe committed:
fix: missing new-line

dviererbe committed:
feat(how-to/get-package-source): Add term reference

dviererbe committed:
fix: spacing issue

dviererbe committed:
chore: Address ambiguous wording

dviererbe committed:
fix: broken doc reference

dviererbe committed:
Merge branch 'overhaul' into add-explanations

Adds RTD config.

dviererbe committed:
fix: sphinx warning "Duplicate explicit target name"

dviererbe committed:
feat: resuse discourse links accross docs

dviererbe committed:
fix: duplicate explicit target name warning

dviererbe committed:
feat: renovate how-to/install-built-packages

dviererbe committed:
feat: renovate how-to/get-package-source

dviererbe commented:
@s-makin
CC: @keirthana 

I am now done (for now) incorporating your feedback. Major changes are:
- Added section "Security update propagation" to explanation/archive
- moved launchpad text markup documentation into reference article
- renovated how-to/install-built-packages
- renovated renovate how-to/get-package-source

dviererbe committed:
Merge branch 'overhaul' into add-explanations

To resolve a merge conflict.

* overhaul:
  chore(how-to/download-new-upstream-version): remove overused term refs
  chore(how-to/download-new-upstream-version): change code to code-block
  fix(how-to/download-new-upstream-version): remove term refs from headings
  chore: convert caution box to section
  chore: Add exaple use cases for uscan
  chore: minor wording improvement
  typo: 'exits' -> 'exists'
  Explain uscan flags --download-version and --download-current-version
  Add back missing WIP banner to how-to/download-new-upstream-version
  Restructered how-to/download-new-upstream-version
  Minor polishing
  Add content for how-to/download-new-upstream-version

s-makin reviewed (approved):
I've looked back through the requested changes and it looks like everything has been addressed. Thanks Dominik!
